src/main/cpp/src/host_table_view.hpp

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/lists/lists_column_view.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <algorithm>
+#include <vector>
+
+namespace spark_rapids_jni {
+
+/**
+ * @brief A non-owning, immutable view of host data as a column of elements,
+ * some of which may be null as indicated by a bitmask.
+ *
+ * Unless otherwise noted, the memory layout of the `host_column_view`'s data and
+ * bitmask is expected to adhere to the Arrow Physical Memory Layout
+ * Specification: https://arrow.apache.org/docs/memory_layout.html
+ *
+ * Because `host_column_view` is non-owning, no host memory is allocated nor freed
+ * when `host_column_view` objects are created or destroyed.
+ */
+class host_column_view {
```

Comment:
I see that this class has almost identical members as `cudf::column_view`. So, given a host data buffer as a column of elements, we should be able to construct an instance of `cudf::column_view` out of it. We just need to keep track that the created instance is viewing host data, and this can be done in Java through a boolean flag.

Did I miss anything here?

Reply:
I thought about using cudf::column_view, but there are member functions that make no sense to call on it (e.g.: the ranged null_count) as it would try to run a device kernel on host data.  Using a separate class makes it type safe, enforced by the compiler, that we aren't passing the wrong type of data to a method.

Reply:
If we want to hide some member functions, we could implement this class with private inheritance of `cudf::column_view`, with explicit exposure of the member functions we want. For example:
```
class host_column_view : cudf::column_view {
public:
  using cudf::column_view::size();
  using cudf::column_view::data();
...
```

Reply:
Does that work in practice given how children work?  cudf::column_view expects the children to be column_views, but they need to be host_column_views in this case.  Seems the closest to what we want is to derive privately from cudf::detail::column_view_base which has everything except the children, and then host_column_view would add just the children interfaces that expect and return host_column_view.  However that would be using a cudf::detail API directly, and given how trivial this stuff is, I'd rather have the duplication than risk breakage given these aren't ever supposed to be used in the same APIs anyway.

Reply:
Okay I see what you meant. Then this would be fine, although that seems not a very good approach when having code duplication.